### PR TITLE
Add `multiple` keyword/functionality to `select`

### PR DESCRIPTION
**Pull request by nimmolo**
This is a reboot of PR #33. It intends to support the [`multiple` HTML spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/multiple) for `<select>` elements.  
```ruby
# Multiple select with automatic [] appending and hidden input
field(:tag_ids).select(
  [1, "Ruby"], [2, "Rails"], [3, "Phlex"],
  multiple: true
)
# Renders: <input type="hidden" name="tag_ids[]" value="">
#          <select name="tag_ids[]" multiple>...</select>
```

- Handles a `multiple` kwarg to the `select` method. Full backwards compatibility with the current API of `select`.
- Field naming: names the `select multiple` field correctly — i.e., using square brackets in a way that does not interfere with parent field notation. (It has to prevent double `[]` when multiple select is used inside collections.) [See this conversation](https://github.com/rubymonolith/superform/pull/33#discussion_r1583413969). Tests all permutations I could think of.
- Adds hidden field when `multiple`, to handle the way browsers handle blank selects (form value may not be submitted unless hidden field present)
- Renames the internal `Select` component param for the array of options from `collection` to `options` (to avoid confusion with `.collection` method, which creates FieldCollections). **Backwards compatible**, both keywords accepted.
- Adds tests for generated HTML.
- Updates the README with examples of the new select API. Also updates the rest of the README with consistent syntax (`field` vs `Field`)
- ~Updates the GEMSPEC to require Ruby 2.7, since the Superform codebase already uses anonymous arg forwarding all over the place (so it already effectively requires 2.7).~

Note, I realize you proposed a separate `multiple_select` component method [in the discussion here] (https://github.com/rubymonolith/superform/pull/33#issuecomment-2073025452) and mentioned that also below, but I would suggest this is not really necessary or desirable, since the current PR maintains backwards compatibility with the current method signature. Keeping the `select` logic in one component seems to be the most intuitive and elegant way. In the `select` HTML element spec, `multiple` is just an attribute, not a separate element.

### CODE REVIEW

**Comment by nimmolo** [comment]
## Add `multiple` and `include_blank` support to Select component

#### Summary

Adds `multiple:` and `include_blank:` keywords to the Select component, renames the `collection:` parameter to `options:` for consistency with Radio component, and fixes value preselection for both single and multiple selects.

## Changes

### Multiple Select Support

```ruby
field(:role_ids).select(
  [[1, "Admin"], [2, "Editor"], [3, "Viewer"]],
  multiple: true
)
```

**Features:**
- Adds `multiple` HTML attribute
- Appends `[]` to field name (`role_ids` → `role_ids[]`)
- Includes hidden input for empty submissions
- **Smart parent detection:** avoids double `[]` when used inside collections

### Include Blank Option

```ruby
field(:country).select(
  [[1, "USA"], [2, "Canada"]],
  include_blank: true
)
```

Renders a **selected blank option** as the first option.

### Parameter Rename: `collection:` → `options:`

Renamed for **semantic clarity** and **consistency** with Radio component:
- `collection:` conflicted with `.collection` method _(creates FieldCollections)_
- `options:` matches HTML semantics _(select contains options)_

**Backwards compatibility:** `collection:` still works with deprecation warning.

### Fixed Value Preselection

**Bug:** Select component did not correctly preselect options based on `field.value`.

**Previous code:**
```ruby
option(selected: field.value == key, value: key) { value }
```

_Failed for multiple selects:_ `[1, 3] == 1` → `false`

**Fixed code:**
```ruby
selected = Array(field.value).include?(key)
option(selected: selected, value: key) { value }
```

Now correctly handles:
- **Single values:** `Array(2).include?(2)` → `true`
- **Array values:** `Array([1, 3]).include?(1)` → `true`
- **Nil values:** `Array(nil).include?(1)` → `false`

## API

### Positional Arguments _(Preferred)_
```ruby
field(:role_ids).select(
  [[1, "Admin"], [2, "Editor"], [3, "Viewer"]],
  multiple: true,
  include_blank: true
)
```

### Options Keyword
```ruby
field(:role_ids).select(
  options: [[1, "Admin"], [2, "Editor"]],
  multiple: true
)
```

### ActiveRecord Relations
```ruby
field(:author_id).select(User.select(:id, :name))
```

## Breaking Changes

**None.** The deprecated `collection:` parameter continues to work with a warning.

## Files Changed

- `lib/superform/rails/components/select.rb` - _Added features and fixed preselection_
- `lib/superform/rails/field.rb` - _Updated helper method_
- `spec/superform/rails/components/select_spec.rb` - _Added 12 new tests (36 total)_
- `README.md` - _Added examples and documentation_

## Test Coverage

**36 tests total** _(24 existing + 12 new)_:
- ✅ Multiple select rendering and field naming
- ✅ Include blank option
- ✅ Collection parent edge cases
- ✅ Single value preselection
- ✅ Array value preselection _(multiple select)_
- ✅ ActiveRecord relation support
- ✅ Backwards compatibility with `collection:` parameter

**All 136 tests passing** _(100 existing + 36 select)_

**Comment by nimmolo** [comment]
Question (also asked on #65, but relevant here)

[In Select currently](https://github.com/rubymonolith/superform/blob/main/lib/superform/rails/components/select.rb), the method and the component accept the "select options" either as positional args, **or sent under the kwarg "collection"**. This feature is undocumented in the README - maybe nobody uses it, and it's effectively dead code?

File: README.md [hunk 1]
```diff
@@ -63,10 +63,10 @@ You probably want to use the same form for creating and editing resources. In Su
 # app/views/posts/form.rb
 class Views::Posts::Form < Components::Form
   def view_template
-    Field(:title).text
-    Field(:body).textarea(rows: 10)
-    Field(:publish_at).date
-    Field(:featured).checkbox
+    field(:title).text
```
Comment (bradgessler) [inline README.md]: These `Field` methods should remain capitalized.

File: lib/superform/rails/components/select.rb [hunk 2]
```diff
@@ -2,22 +2,55 @@ module Superform
...
+          @options = options
+          @multiple = multiple
         end
 
-        def view_template(&options)
+        def view_template(&block)
+          # Hidden input ensures a value is sent even when all options are
+          # deselected in a multiple select
+          if @multiple
+            hidden_name = field.parent.is_a?(Superform::Field) ? dom.name : "#{dom.name}[]"
```
Comment (bradgessler) [inline lib/superform/rails/components/select.rb]: I think I use `"#{dom.name}[]"` in a collection somewhere. Could that same class be used here? If it doesn't make sense to drop that class in here then `"#{dom.name}[]"` should be dropped into a helper class method somewhere and called so it goes through the same code path.

File: lib/superform/rails/components/select.rb [hunk 3]
```diff
@@ -2,22 +2,55 @@ module Superform
...
+              options(*filtered_options)
+            end
           end
         end
 
         def options(*collection)
           map_options(collection).each do |key, value|
-            option(selected: field.value == key, value: key) { value }
+            # Handle both single values and arrays (for multiple selects)
+            selected = Array(field.value).include?(key)
```
Comment (bradgessler) [inline lib/superform/rails/components/select.rb]: This allocates a new array per iteration. Move `selected` outside of the loop if it doesn't need to be there and then check if the key is included in `selected`.

**Review by bradgessler — CHANGES_REQUESTED** [review]
This commit feels "too big" for the functionality it implements. Changes to the README, 500+ LOC specs, and the Ruby version change feels like an LLM got carried away.

As far as the actual code goes, I left comments inline for a few things to change.

File: superform.gemspec [hunk 4]
```diff
@@ -12,7 +12,7 @@ Gem::Specification.new do |spec|
   spec.description = "A better way to customize and build forms for your Rails application"
   spec.homepage = "https://github.com/rubymonolith/superform"
   spec.license = "MIT"
-  spec.required_ruby_version = ">= 2.6.0"
+  spec.required_ruby_version = ">= 2.7.0"
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply superform.gemspec]: Reverting the required Ruby version bump on this branch, moving it to a PR branch.

File: lib/superform/rails/components/select.rb (hunk 3 above)
Comment (nimmolo) [reply lib/superform/rails/components/select.rb]: Done

File: lib/superform/rails/field.rb [hunk 5]
```diff
@@ -45,8 +45,17 @@ def textarea(**attributes)
         Components::Textarea.new(field, attributes:)
       end
 
-      def select(*collection, **attributes, &)
-        Components::Select.new(field, attributes:, collection:, &)
+      def select(*options, **attributes, &)
+        # Extract select-specific parameters from attributes
+        multiple = attributes.delete(:multiple) || false
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply lib/superform/rails/field.rb]: Understood - done.

File: lib/superform/rails/components/select.rb (hunk 2 above)
Comment (nimmolo) [reply lib/superform/rails/components/select.rb]: I see what you mean. This is something we probably want to be reusable, in light of the checkbox/radio group PR coming up. 

Does it make sense to add a method to the DOM class called `array_name`?
```ruby
# Returns the name with `[]` appended for array/multiple value fields.
# Used by multiple selects, checkbox groups, etc.
def array_name
  "#{name}[]"
end
```
